src/daemon/plugin.c

```diff
@@ -578,10 +578,16 @@ static void *plugin_read_thread(void __attribute__((unused)) * args) {
 
     /* Check, if `rf_next_read' is in the past. */
     if (rf->rf_next_read < now) {
-      /* `rf_next_read' is in the past. Insert `now'
-       * so this value doesn't trail off into the
-       * past too much. */
-      rf->rf_next_read = now;
+      if (rf->rf_ctx.align_read < 0) {
+        /* `rf_next_read' is in the past. Insert `now' so this value doesn't
+         * trail off into the past too much. */
+        rf->rf_next_read = now;
+      } else {
+        /* ensure read alignment, if enabled */
+        while (rf->rf_next_read < now) {
```

Comment:
You remove one loop and put another instead it?

Facepalm.

Reply:
Yes, because the loop body is very likely only executed once. An adequate expression would introduce more overhead in most cases due to the necessary division.

Reply:
Sounds reasonable..

Reply:
Please add comment about "previous rf_next_read value is aligned". That is unclear here.

Another point, where alignment could be lost, is:

`rf->rf_effective_interval = max_read_interval;` + `rf->rf_next_read += rf->rf_effective_interval;`

Reply:
I changed the comment and added a check for the _max_read_interval_ value.